go/internal/feast/featurestore.go

```
@@ -76,6 +77,75 @@ func NewFeatureStore(config *registry.RepoConfig, callback transformation.Transf
 	}, nil
 }
 
+func entityTypeConversion(entityMap map[string]*prototypes.RepeatedValue, entityColumns map[string]*model.Field) (map[string]*prototypes.RepeatedValue, error) {
+	for entityName, entityValue := range entityMap {
+		if entityColumn, ok := entityColumns[entityName]; ok {
+			newEntityValue := &prototypes.RepeatedValue{}
+			for _, value := range entityValue.Val {
+				newVal, err := types.ConvertToValueType(value, entityColumn.Dtype)
+				if err != nil {
+					return nil, fmt.Errorf("error converting entity value for %s: %w", entityName, err)
+				}
+				newEntityValue.Val = append(newEntityValue.Val, newVal)
+			}
+			entityMap[entityName] = newEntityValue
+		} else {
+			return nil, fmt.Errorf("entity %s not found in entity columns", entityName)
+		}
```

Clear error message here or Ignore this error. This validation conflicts with serving.ValidateEntityValues() as well. For now extra entities passed are ignored in ValidateEntityValues. We can change it so all validation happen at one place.